python/ray/dataframe/dataframe.py

```diff
@@ -467,14 +469,10 @@ def _correct_dtypes(self):
         Args:
             block_partitions: arglist of column blocks.
         """
-        if self._block_partitions.shape[0] > 1:
-            self._block_partitions = np.array(
-                    [_correct_column_dtypes._submit(
-                     args=column, num_return_vals=len(column))
-                     for column in self._block_partitions.T]).T
-
-        self._dtypes_cache = [_deploy_func.remote(lambda df: df.dtypes, pd_df)
-                              for pd_df in self._block_partitions[0]]
+        small_dfs = [_map_partitions(lambda df: df.loc[0:0], column)
```

Comment:
Pandas has some useful library functions: `pandas/core/dtypes/cast.py`, which can coalesce and upcast dypes as needed. It might be useful to investigate these to see if we can use their rules directly to get dtypes.

Reply:
On further investigation, the overhead of df.loc[0:0] and concatenating these along columns is not much, and so might just be better to combine the remote tasks to work on column aligned blocks. And then do the `ray.get` and `concat` on the driver in the getter method, as discussed.